### PR TITLE
Setup wizard: Omit PayPal Standard unless plugins cannot be installed

### PR DESCRIPTION
**Pull request by dechov**
Fixes: https://github.com/woocommerce/woocommerce/issues/17374

Excludes PayPal Standard from available gateways:

Before | After
-- | --
 | 

...unless the user lacks permission to install plugins:

<s>This makes the `get_wizard_in_cart_payment_gateways` function slightly shorter but some further refactoring of that function (as [requested](https://github.com/woocommerce/woocommerce/issues/17374#issuecomment-351139584)) is still in scope here.</s>
The `get_wizard_in_cart_payment_gateways` function was refactored, extracting a function that returns all possible in-cart gateways, and generally tightening the logic.

Also, a few minor changes:
- Make PPEC opt-out in every case (restore opt-out change from https://github.com/woocommerce/woocommerce/pull/18040)
- Fix 'Klarna Payments' name (rendered as alt text)
- Harden logic by checking Stripe support before adding it in Klarna and Square cases

Summary of the logic, for testing purposes:
- If plugins can't be installed, just show PayPal Standard [disabled]
- Otherwise, if Klarna Checkout, Klarna Payments, or Square supported in country, show it [enabled], followed by PPEC [enabled], followed by Stripe [disabled]
- Otherwise, show Stripe [enabled] (if supported) and PPEC [enabled]

### CODE REVIEW

File: includes/admin/class-wc-admin-setup-wizard.php [hunk 1]
```diff
@@ -1108,139 +1103,139 @@ protected function get_wizard_in_cart_payment_gateways() {
...
+						'value'       => 'yes',
+						'placeholder' => '',
+						'required'    => false,
+					),
+					'email' => array(
+						'label'       => __( 'Stripe email address:', 'woocommerce' ),
+						'type'        => 'email',
+						'value'       => $user_email,
+						'placeholder' => __( 'Stripe email address', 'woocommerce' ),
+						'description' => __( "Enter your email address and we'll handle account creation. WooCommerce Services and Jetpack will be installed and activated for you.", 'woocommerce' ),
```
Comment (kloon) [inline includes/admin/class-wc-admin-setup-wizard.php]: Nitpick, but single quotes should be used here.

File: includes/admin/class-wc-admin-setup-wizard.php [hunk 2]
```diff
@@ -1108,139 +1103,139 @@ protected function get_wizard_in_cart_payment_gateways() {
...
+						'value'       => 'yes',
+						'placeholder' => '',
+						'required'    => false,
+					),
+					'api_subject' => array(
+						'label'       => __( 'Direct payments to email address:', 'woocommerce' ),
+						'type'        => 'email',
+						'value'       => $user_email,
+						'placeholder' => __( 'Email address to receive payments', 'woocommerce' ),
+						'description' => __( "Enter your email address and we'll authenticate payments for you. To claim a payment, you'll need to have a PayPal Business account or create one later. WooCommerce Services and Jetpack will be installed and activated for you.", 'woocommerce' ),
```
Comment (kloon) [inline includes/admin/class-wc-admin-setup-wizard.php]: Single quotes instead of double quotes.

**Comment by kloon** [comment]
Looks good @dechov, just some small nitpicks. Also would it be possible to perhaps add a couple of unit tests to test the country scenarios? Just to make sure certain gateways are only shows when the base country is a certain one?

File: includes/admin/class-wc-admin-setup-wizard.php (hunk 1 above)
Comment (mikejolley) [reply includes/admin/class-wc-admin-setup-wizard.php]: String contains single quote so ok.

File: includes/admin/class-wc-admin-setup-wizard.php (hunk 2 above)
Comment (mikejolley) [reply includes/admin/class-wc-admin-setup-wizard.php]: String contains single quote so ok.

**Comment by dechov** [comment]
👍 Added a test asserting on example countries, but let me know if there's any more robust way to "make sure certain gateways only show when the base country is a certain one".

Also fixed Square support in the UK.

**Pull request merged**